### PR TITLE
Improve configuration for digital twin adapters and mock

### LINKED ISSUE #17: [enhancement] Improve Configuration at Runtime
## Description

~~Improve application configuration so that there is a single application-wide configuration file rather than requiring multiple configs in various locations to be managed. In addition, consider providing both a template config file and a working sample or default configuration.~~

~~Since many configuration pieces will be related to pluggable components, it would also be acceptable to have a "main" file which references other configuration files.~~

~~In addition, it is probably best to switch to YAML files to mirror other SDV components~~

Improve the sample adapter configuration model to enable easier configuration at runtime. Currently, the adapters would require changes to source code or files in non-static build directories to update configuration. These components should search some set of known locations for their configuration so that they can be more easily configured at runtime. In particular, this should support the Ibeji examples and SDV blueprint scenarios, allowing them to use the sample adapters with custom configuration.

## Acceptance criteria

- ~~Application uses only a single config file OR there is a "main" configuration file that references component sub-configurations~~
- ~~A template is available for users to customize~~
- ~~A working sample or default file is available for users who want to get started quickly~~
- ~~Configuration file(s) is/are in YAML format~~
- [x] Sample mapping client adapters support external config
- [ ] Sample digital twin adapters support external config
- [x] ~~Sample cloud adapters support external config~~ The sample doesn't require configuration, so the configuration was removed entirely
- [x] Provider proxies support external config (which would solve #50)

### PR DESCRIPTION
**Pull request by wilyle**
Improve configuration for digital twin adapters and mock

Closes #17

### CODE REVIEW

File: mocks/mock_digital_twin/README.md [hunk 1]
```diff
@@ -4,12 +4,12 @@ The Mock Digital Twin mocks the behavior of the in-vehicle digital twin services
...
 
-The mock can be configured via the `res/config.json` file which is copied to the build output automatically. This file contains the following properties:
+This mock supports the following configuration:
 
 - `digital_twin_server_authority`: the authority that will be used for hosting the mock digital twin service
-- `config_items`: a list of objects with the following properties:
-  - `begin`: an integer indicating when to enable the entity below
-  - `end`: an optional integer indicating when to disable the entity below. Set to `null` if you never want the entity to "turn off"
+- `entities`: a list of entities with the following properties:
+  - `begin`: an integer indicating when to enable this entity
```
Comment (ashbeitz) [inline mocks/mock_digital_twin/README.md]: An example for begin and end would be helpful.

**Review by ashbeitz — COMMENTED** [review]
One comment.  Please take a look.

File: mocks/mock_digital_twin/README.md (hunk 1 above)
Comment (wilyle) [reply mocks/mock_digital_twin/README.md]: there are example values in the default config and the behavior section discusses these properties in more detail, so I will just mention that section here

**Review by ashbeitz — APPROVED** [review]
Approved.

**Pull request merged**